### PR TITLE
flexbox: add component

### PR DESCRIPTION
**Pull request by uselessdev**
This PR add the `Flexbox` component. 
With this component we dont't need to keep repeating pieces of code that are styles that only add flexbox.

Interface:
``` jsx
<Flexbox
  alignItems="flex-start"
  justifyContent="flex-start"
  direction="row"
>
  {children}
</Flexbox>
```

The `alignItems` prop can receive: `flex-start` (default), `center`, `flex-end`, `stretch` and `baseline`
The `justifyContent` prop can receive: `flex-start` (default), `center`, `flex-end`, `space-between`, `space-around` and `space-evenly`.
The `direction` prop can receive: `row` (default), `column`, `row-reverse` and `column-reverse`.

Depends from pagarme/former-kit-skin-pagarme#115

### CODE REVIEW

File: src/Box/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+const consumeTheme = ThemeConsumer('UIBox')
+
+const Box = ({
+  align,
+  children,
+  className,
+  justify,
+  theme,
+}) => (
+  <div className={classNames(theme.box, className, {
```
Comment (LeoNero) [inline src/Box/index.js]: O que acha de colocar className numa linha separada?

File: src/Box/index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      [theme[align]]: align,
+      [theme[justify]]: justify,
+    })}
+  >
+    {children}
+  </div>
+)
+
+Box.propTypes = {
+  align: PropTypes.oneOf([
```
Comment (LeoNero) [inline src/Box/index.js]: :warning: Documentation!

File: src/Box/index.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    'middle',
+    'bottom',
+    'top',
+    'stretch',
+  ]),
+  children: PropTypes.node.isRequired,
```
Comment (LeoNero) [inline src/Box/index.js]: :warning: Documentation!

File: src/Box/index.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  className: PropTypes.string,
```
Comment (LeoNero) [inline src/Box/index.js]: :warning: Documentation!

File: src/Box/index.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  justify: PropTypes.oneOf([
```
Comment (LeoNero) [inline src/Box/index.js]: :warning: Documentation!

File: src/Box/index.js (hunk 1 above)
Comment (LeoNero) [inline src/Box/index.js]: What do you think about putting `classNames` in the line below `<div`?

File: src/Box/index.js (hunk 2 above)
Comment (uselessdev) [reply src/Box/index.js]: Thanks @LeoNero I don't documented the props yet, because I want to discuss if this component is really necessary.

**Comment by uselessdev** [comment]
@Deivis já tem `Spacing` ai mas olha essa issue aqui: pagarme/former-kit-skin-pagarme#105

**Comment by Deivis** [comment]
Agora q vi :disappointed:

File: src/index.js [hunk 6]
```diff
@@ -1,5 +1,6 @@
 export { default as Alert } from './Alert'
 export { default as Avatar } from './Avatar'
+export { default as Box } from './Box'
```
Comment (rsmelo) [inline src/index.js]: ajustar esse import

File: config/styleguide.config.js [hunk 7]
```diff
@@ -19,6 +19,7 @@ module.exports = {
       components: () => [
         '../src/Alert/index.js',
         '../src/Avatar/index.js',
+        '../src/Box/index.js',
```
Comment (Deivis) [inline config/styleguide.config.js]: E este import de box  ?

File: src/Flexbox/index.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#### **Examples** ####
+
+Default
+```jsx
+const Button = require('../Button').default;
+const Spacing = require('../Spacing').default;
+
+<Box align="top" justify="start">
```
Comment (rsmelo) [inline src/Flexbox/index.md]: need to change `Box` to `Flexbox`

File: src/Flexbox/index.js [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+Flexbox.propTypes = {
+  /**
+   * The align-items props
+   */
+  alignItems: PropTypes.oneOf([
+    'baseline',
+    'flex-end',
+    'center',
+    'stretch',
+    'flex-start',
```
Comment (Deivis) [inline src/Flexbox/index.js]: alphabetical order

File: src/Flexbox/index.js [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+      [theme.top]: alignItems === 'flex-start',
+      [theme.stretch]: alignItems === 'stretch',
+      [theme.around]: justifyContent === 'space-around',
+      [theme.between]: justifyContent === 'space-between',
+      [theme.evenly]: justifyContent === 'space-evenly',
+      [theme.center]: justifyContent === 'center',
+      [theme.end]: justifyContent === 'flex-end',
+      [theme.start]: justifyContent === 'flex-start',
+      [theme.row]: direction === 'row',
+      [theme.column]: direction === 'column',
```
Comment (Deivis) [inline src/Flexbox/index.js]: alphabetical order

File: src/Flexbox/index.js [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+  ]),
+  /**
+   * How childrens will be distribuited.
+   */
+  justifyContent: PropTypes.oneOf([
+    'space-around',
+    'space-between',
+    'center',
+    'flex-end',
+    'space-evenly',
```
Comment (Deivis) [inline src/Flexbox/index.js]: alphabetical order

File: src/Flexcontainer/index.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
+#### **Examples** ####
+
+Default
+```jsx
+const Button = require('../Button').default;
+const Spacing = require('../Spacing').default;
+
+<Flexcontainer align="top" justify="start">
```
Comment (rsmelo) [inline src/Flexcontainer/index.md]: `<Flexcontainer alignItems="top" justifyContent="start">`

File: src/index.js [hunk 13]
```diff
@@ -8,6 +8,7 @@ export { default as CheckboxGroup } from './CheckboxGroup'
 export { default as DateInput } from './DateInput'
 export { default as DateSelector } from './DateSelector'
 export { default as Dropdown } from './Dropdown'
+export { default as Flexbox } from './Flexbox'
```
Comment (Deivis) [inline src/index.js]: Wrong import

File: stories/index.js [hunk 14]
```diff
@@ -10,6 +10,7 @@ import './Colors'
 import './DateInput'
 import './DateSelector'
 import './Dropdown'
+import './Flexbox'
```
Comment (Deivis) [inline stories/index.js]: Wrong import

File: src/Flexbox/index.js [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+      [theme.column]: direction === 'column',
+      [theme.columnReverse]: direction === 'column-reverse',
+      [theme.end]: justifyContent === 'flex-end',
+      [theme.evenly]: justifyContent === 'space-evenly',
+      [theme.middle]: alignItems === 'center',
+      [theme.row]: direction === 'row',
+      [theme.rowReverse]: direction === 'row-reverse',
+      [theme.start]: justifyContent === 'flex-start',
+      [theme.stretch]: alignItems === 'stretch',
+      [theme.top]: alignItems === 'flex-start',
```
Comment (lucasjs) [inline src/Flexbox/index.js]: Melhor colocar os nomes reais, pois quando muda o direction, o top, bottom, etc mudam o sentido.

Comment (oieusouamiguel) [reply src/Flexbox/index.js]: concordo com o @lucasjs

**Comment by lucasjs** [comment]
Que tal os dois ultimos containers terem fundo cinza também?

image

**Comment by lucasjs** [comment]
@uselessdev muito bom! Só faltou o `inherit`.

**Comment by lucasjs** [comment]
Fechou.

**Pull request merged**